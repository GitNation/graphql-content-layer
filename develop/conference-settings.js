const fallback = require('./fallback-settings');

const confCodes = [
  'jsn',
  'gqconf',
  'mlconf',
  'qaconf',
  'doconf',
  'nodeconf',
  'rs',
];

const getSettings = () => {
  const settings = confCodes.reduce(
    (acc, code) => ({ ...acc, [code]: fallback }),
    {},
  );
  try {
    confCodes.forEach(code => {
      settings[
        code
      ] = require(`../../live-conferences/src/conferences/${code}/conference-settings`);
    });
    return settings;
  } catch (err) {
    if (err.code === 'MODULE_NOT_FOUND') {
      console.log(
        `\n\nCan't find live-conferences folder. Fallback settings will be used!\n\n`,
      );
      return settings;
    }
    throw err;
  }
};

module.exports = { getSettings };
