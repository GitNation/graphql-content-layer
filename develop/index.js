const fs = require('fs');
const path = require('path');

const { getContent } = require('../dist');
const { getSettings } = require('./conference-settings');

const CURRENT_CONF = 'rs';

const testLaunch = async () => {
  const settings = getSettings();
  const content = await getContent(settings[CURRENT_CONF]);

  fs.writeFileSync(
    path.resolve(__dirname, '../content-log.json'),
    JSON.stringify(content, null, 2),
  );
  // console.log('content', content);
};

testLaunch();
