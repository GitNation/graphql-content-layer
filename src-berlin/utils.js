const { labelColors } = require('./config');

const labelTag = prefix => label => {
  const colorInfo =
    labelColors.find(c => c.label === label) ||
    labelColors.find(c => c.label === null);
  return `${prefix}--${colorInfo.tag}`;
};

module.exports = {
  labelTag
};
