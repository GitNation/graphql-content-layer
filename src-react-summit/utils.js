const { markdownToHtml } = require('./markdown');

const prepareSpeakers = async speakers => {
  const allSpeakersAsync = speakers
    .filter(Boolean)
    .map(item => ({
      ...item.speaker,
      ...item,
      avatar: (item.speaker && item.speaker.avatar) || item.avatar || {},
    }))
    .map(
      async ({
        bio,
        githubUrl,
        twitterUrl,
        mediumUrl,
        ownSite,
        speaker,
        avatar,
        ...item
      }) => ({
        ...item,
        company: `${item.company}, ${item.country}`,
        photo: avatar && avatar.url,
        desc: await markdownToHtml(bio),
        github: githubUrl,
        twitter: twitterUrl,
        medium: mediumUrl,
        site: ownSite,
      })
    );
  const allSpeakers = await Promise.all(allSpeakersAsync)
  const uniqueSpeakers = new Set(allSpeakers.map(({ name }) => name));
  const readySpeakers = [...uniqueSpeakers].map(name => allSpeakers.find(speaker => speaker.name === name));
  return readySpeakers;
};

module.exports = {
  prepareSpeakers,
};
