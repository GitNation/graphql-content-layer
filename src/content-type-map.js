const contentTypeMap = {
  AdviceLounge: '31ae96063a25477f8f007379231c1482',
  ConferenceBrand: '2d89f139f3d042ab98d212062527eb82',
  ConferenceEvent: '20e4c3e5323f41cba1ae8840df6e6d32',
  DaySchedule: '2e92d35e0ac0421baf779aef56eec40d',
  Diversity: '55eb254b4ccb4cb1a4c380b7aaa7dc17',
  Extended: 'c20fac4698f94d148fcbb27989ef48e4',
  Faq: '4f4e1bf4ce884d6e99735af934488515',
  Job: '41e965599f914b84947ce04d901dc088',
  Mc: '29d69d9b48494ef38ac0d7a22012eeab',
  Page: '371641c28b044b00bf532b2dd6e1516d',
  PieceOfSpeakerInfo: 'd31070ab4611456a83fa837a6ccea527',
  PieceOfSponsorInfo: 'c52cf3fb6f1c41588ea7fc77a092f419',
  PieceOfText: 'c04bfcdd2eed47b880588f5506ed4f5a',
  Speaker: 'e981e312fd6c48a39230de9193be4db9',
  Sponsor: 'f188bcf25c8e419d9f8d10b0926dd825',
  // Talk: 'd5d4bfeb605c41569b30d478a03bbb0e',
  Track: '76d236bfe3344d0990028c5f69c69079',
  Workshop: '2ce17eb4016e4be5ae72bb0db484f3f0',

  // updated
  OrgEvent: 'cda52433c9fd4a6a818f71acd217e91d',
  Talk: '03b4712293184731b7d1bf074302f8e0',
  QA: '3473d8bc2712475c89e0b47d31b2ee97',
  LightningTalk: '09be1744ee4d465ea3f5a82972d2cb28',
  GroupLT: '694d3ca13ab44d45810c18fb01580c92',
  SpeakersRoom: 'b701d52be9d84599a3d46972acafc42f',
  DiscussionRoom: '07f1cfefde1942fbb63704917cbf2398',

};

module.exports = {
  contentTypeMap,
};
