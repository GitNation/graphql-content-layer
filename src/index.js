const { GraphQLClient } = require('graphql-request');

const { credentials, conferenceTitle, eventYear } = require('./config');
const textContent = require('./fetch-texts');
const pageContent = require('./fetch-pages');
const brandContent = require('./fetch-brand');
const speakerContent = require('./fetch-speakers');
const performanceContent = require('./fetch-performance');
const sponsorContent = require('./fetch-sponsors');
const talksContent = require('./fetch-talks');
const workshopContent = require('./fetch-workshops');
const mcContent = require('./fetch-mc');
const faqContent = require('./fetch-faq');
const extContent = require('./fetch-extended');
const jobsContent = require('./fetch-jobs');
const committeeContent = require('./fetch-committee');

const createClient = ({ endpoint, token }) => {
  return new GraphQLClient(endpoint, {
    headers: {
      authorization: `Bearer ${token}`,
    },
  });
};

const client = createClient(credentials);
const queriesData = [];

const getQueriesData = content => {
  try {
    const { fetchData, ...data } = content;
    if (!Object.keys(data).length) return;
    queriesData.push(data);
  } catch (err) {}
};

const getContent = async () => {
  const fetchAll = [
    textContent,
    pageContent,
    brandContent,
    speakerContent,
    performanceContent,
    sponsorContent,
    talksContent,
    workshopContent,
    mcContent,
    faqContent,
    extContent,
    jobsContent,
    committeeContent,
  ].map(async content => {
    try {
      getQueriesData(content);
      return await content.fetchData(client, { conferenceTitle, eventYear });
    } catch (err) {
      console.error(err);
    }
  });

  const contentArray = await Promise.all(fetchAll);
  const contentMap = contentArray.reduce((content, piece) => {
    try {
      const newKeys = Object.keys(piece);
      const existentKeys = Object.keys(content);
      const intersectedKeys = newKeys.filter(k => existentKeys.includes(k));
      intersectedKeys.forEach(k => {
        piece[k] = { ...content[k], ...piece[k] };
      });
    } catch (err) {}
    return { ...content, ...piece };
  }, {});
  return contentMap;
};

module.exports = {
  getContent,
  queriesData,
};
