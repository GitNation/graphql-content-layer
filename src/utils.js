const { markdownToHtml } = require('./markdown');

const tagColors = {
  'NodeJS': {
    tagBG: '#7AB464',
    color: '#fff',
  },
  'WebGL': {
    tagBG: '#ff7302',
    color: '#ffffff'
  },
  'Ecosystem': {
    tagBG: '#fff40d',
    color: '#ae4f01'
  },
  'Language': {
    tagBG: '#fff40d',
    color: '#ae4f01'
  },
  'GraphQL': {
    tagBG: '#f200fa',
    color: '#400042',
  },
  'VueJS': {
    tagBG: '#4EBA87',
    color: '#fff',
  },
  'Performance': {
    tagBG: '#00ed24',
    color: '#00410a',
  },
  'TypeScript': {
    tagBG: '#61DAFB',
    color: '#030303'
  },
  default: {
    tagBG: 'black',
    color: 'white'
  }
};

const getSocials = speaker => {
  const ICONS = {
    githubUrl: 'gh',
    twitterUrl: 'tw',
    mediumUrl: 'med',
    ownSite: 'site',
  };
  const { githubUrl, twitterUrl, mediumUrl, ownSite, companySite } = speaker;
  const socials = Object.entries({ githubUrl, twitterUrl, mediumUrl, ownSite, companySite })
    .map(([key, val]) => (val && { link: val, icon: ICONS[key] }))
    .filter(Boolean);
  return socials;
};

const getLabelColor = label => {
  const colors = tagColors[label] || tagColors.default;
  return colors;
};

const prepareSpeakers = speakers => speakers
  .map(item => ({
    ...item.speaker,
    ...item,
    avatar: item.speaker.avatar || {},
  }))
  .map(
    async ({
      bio,
      speaker,
      avatar,
      ...item
    }) => ({
      ...item,
      company: `${item.company}, ${item.country}`,
      avatar: avatar.url,
      bio: await markdownToHtml(bio),
      socials: getSocials(item),
      ...getLabelColor(item.label),
    })
  );

module.exports = {
  getLabelColor,
  prepareSpeakers,
  tagColors,
};
