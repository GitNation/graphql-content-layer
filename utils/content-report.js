import hash from 'object-hash';

const createPagesReport = pages => {
  const pagesKeys = Object.keys(pages);
  const report = pagesKeys.map(key => {
    const page = pages[key];
    const {
      id,
      titleSeo,
      description,
      seoDescription,
      titlePage,
      ...rest
    } = page;
    return {
      pageCode: key,
      id,
      titleSeo,
      description,
      seoDescription,
      titlePage,
      content: hash(rest),
    };
  });
  return report;
};

export const createReport = content => {
  try {
    const {
      pages,
      conference,
      speakers,
      sponsors,
      schedule,
      tracks,
      talks,
      workshops,
      ...rest
    } = content;
    return {
      pages: createPagesReport(pages),
      conference,
    };
  } catch (err) {
    console.error(err);
    return {};
  }
};
