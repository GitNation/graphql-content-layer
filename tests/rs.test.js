import { getSettings } from '../develop/conference-settings';
import { getContent } from '../src';
import { createReport } from '../utils/content-report';

describe('RS', () => {
  const settings = getSettings();

  it.each([
    'pages',
    'conference',
    'speakers',
    'sponsors',
    'schedule',
    'tracks',
    'talks',
    'workshops',
    'otherContent',
  ])('should prepare %s', async contentSection => {
    const content = await getContent(settings.rs);
    const report = createReport(content);
    expect(report[contentSection]).toMatchSnapshot();
  });
});
