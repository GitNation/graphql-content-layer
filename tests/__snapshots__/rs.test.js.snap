// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RS should prepare conference 1`] = `
Object {
  "city": "Online, 2 days",
  "codeOfConductIntro": "<p>All attendees, speakers, sponsors and volunteers at React Summit Conference and React Amsterdam Meetups are required to follow this Code of Conduct. Organisers will strictly enforce it throughout the events and community gatherings. We expect cooperation from all participants to help ensure a safe environment for everybody.</p>
<p>In case you have identified a breach of our CoC, please approach our crew (organizers and volunteers in branded t-shirts) immediately or contact our CoC enforcement team - <a href=\\"mailto:coc@gitnation.org\\">coc@gitnation.org</a>, +44 7376 237 299.</p>
",
  "codeOfConductMain": "<h2>The Short Version</h2>
<p>Our conference is dedicated to providing a harassment-free conference experience for everyone, regardless of gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, ethnicity, religion (or lack thereof), or technology choices. We do not tolerate harassment of conference participants in any form. Sexual language and imagery is not appropriate for any conference venue, including talks, workshops, parties, Twitter and other online media. Conference participants violating these rules may be sanctioned or expelled from the conference without a refund at the discretion of the conference organisers.</p>
<h2>The Long Version</h2>
<p>Harassment includes offensive verbal comments related to gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, ethnicity, religion, technology choices, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention.</p>
<p>Participants asked to stop any harassing behavior are expected to comply immediately.</p>
<p>Sponsors are also subject to the anti-harassment policy. In particular, sponsors should not use sexualised images, activities, or other material. Booth staff (including volunteers) should not use sexualised clothing/uniforms/costumes, or otherwise create a sexualised environment.</p>
<p>If a participant engages in harassing behavior, the conference organisers may take any action they deem appropriate, including warning the offender or expulsion from the conference with no refund.</p>
<p>If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact a member of conference staff immediately. Conference staff can be identified as they'll be wearing branded clothing and/or badges.</p>
<p>Conference staff will be happy to help participants contact hotel/venue security or local law enforcement, provide escorts, or otherwise assist those experiencing harassment to feel safe for the duration of the conference. We value your attendance.</p>
<p>We expect participants to follow these rules at conference and workshop venues and conference-related social events.</p>
<p>Original source and credit: <a href=\\"http://2012.jsconf.us/#/about\\">http://2012.jsconf.us/#/about</a> &#x26; The Ada Initiative</p>
<p>Please help by translating or improving: <a href=\\"https://github.com/leftlogic/confcodeofconduct.com\\">http://github.com/leftlogic/confcodeofconduct.com</a></p>
<p>This work is licensed under a Creative Commons Attribution 3.0 Unported License</p>
",
  "contentType": "2d89f139f3d042ab98d212062527eb82",
  "facebookUrl": "https://www.facebook.com/reactamsterdam",
  "id": "ck0zi93n8phx10b853erl9psf",
  "mediumUrl": "https://medium.com/@ReactAmsterdam",
  "slackUrl": "https://reactsummit.com/slack",
  "status": "PUBLISHED",
  "title": "React_Amsterdam",
  "twitterUrl": "https://twitter.com/reactsummit",
  "url": "https://reactsummit.com",
  "youtubeUrl": "https://youtube.com/c/ReactConferences",
}
`;

exports[`RS should prepare otherContent 1`] = `undefined`;

exports[`RS should prepare pages 1`] = `
Array [
  Object {
    "content": "d94c53bd40d0306cbb9d374211e738e1fe2061a3",
    "description": "October 15-16, 2020",
    "id": "ck13wjjma8piw0b49xozji0jy",
    "pageCode": "main",
    "seoDescription": "React Summit is two-day two-track conference of all things React, gathering Front-end developers across the globe in the cloud.",
    "titlePage": "THE BIGGEST REACT CONFERENCE WORLDWIDE",
    "titleSeo": "React Summit â€“ The Biggest React Conference Worldwide",
  },
  Object {
    "content": "a56cb0520d2a056b63604d9796a63513ab631743",
    "description": "October 15-16, 2020",
    "id": "ck1821481qob90b49rwasqgcn",
    "pageCode": "coc",
    "seoDescription": "React Summit is a 2-day conference of all things React, gathering Front-end developers across the globe.",
    "titlePage": "Code of Conduct",
    "titleSeo": "Code of Conduct at React Summit",
  },
  Object {
    "content": "5a4d4507fde29c6beb02df703cd1f85751378c42",
    "description": "You are a React developer and would love to work for a company supporting React community and getting technology stack serious?",
    "id": "ck182onb491lg0b71gwe31knv",
    "pageCode": "jobs",
    "seoDescription": "React Summit is a full day two-track conference of all things React, gathering Front End developers across the globe in the tech heart of Europe.",
    "titlePage": "Top React Jobs",
    "titleSeo": "React Jobs from Most Notable React Community Supporters",
  },
  Object {
    "content": "5e567b65df275f6219684711fa969bf630d0c9a6",
    "description": "React Summit is not just a big conference. It is a social and cultural environment designed to fit React crowd in the way engineers like with an extra addition to you regular conference experience.",
    "id": "ck183tplbqvzb0b85upjvy1fe",
    "pageCode": "workshops",
    "seoDescription": "React Summit is a full day two-track conference of all things React, gathering Front-end developers across the globe in the tech heart of Europe.",
    "titlePage": "LEARN FROM THE BEST TRAINERS IN THE CLOUD",
    "titleSeo": "Workshops at React Summit ",
  },
  Object {
    "content": "16807dfe9f26906aba2c460c930ab6f6b1671a0e",
    "description": "October 15-16, 2020",
    "id": "ck3okkfzitf3q0b662wyy68k9",
    "pageCode": "FAQ",
    "seoDescription": "React Summit is a full day two-track conference of all things React, gathering Front-end developers across the globe in the tech heart of Europe.",
    "titlePage": "FAQ",
    "titleSeo": "FAQ",
  },
  Object {
    "content": "9bfddce65d0c8cc586e56a29f85aa59ec0af6436",
    "description": "React Summit is not just a big conference. It is a social and cultural environment designed to fit React crowd in the way engineers like with an extra addition to you regular conference experience.",
    "id": "cke0tievc0tfh0a85yn4bbhav",
    "pageCode": "workshops_alt",
    "seoDescription": "React Summit is a full day two-track conference of all things React, gathering Front-end developers across the globe in the tech heart of Europe.",
    "titlePage": "LEARN FROM THE BEST TRAINERS IN THE CLOUD",
    "titleSeo": "Workshops at React Summit ",
  },
  Object {
    "content": "8b11774031e188f5518b716163018bbd67156137",
    "description": null,
    "id": "ckf5mcb0sbctl0b77sent6ar5",
    "pageCode": "perks",
    "seoDescription": null,
    "titlePage": "Special Offers from Sponsors",
    "titleSeo": "Special Offers from Sponsors",
  },
]
`;

exports[`RS should prepare schedule 1`] = `undefined`;

exports[`RS should prepare speakers 1`] = `undefined`;

exports[`RS should prepare sponsors 1`] = `undefined`;

exports[`RS should prepare talks 1`] = `undefined`;

exports[`RS should prepare tracks 1`] = `undefined`;

exports[`RS should prepare workshops 1`] = `undefined`;
